### PR TITLE
PROD-1889, PROD-1899, PROD-1920, PROD-1902, PROD-2127, PROD-1922, PROD-1921, PROD-2106 - Custom Profile & Group Layouts

### PR DESCRIPTION
**Pull request by krunal-buddyboss**
### Jira Issue:
https://buddyboss.atlassian.net/browse/PROD-1988

https://buddyboss.atlassian.net/browse/PROD-1889
https://buddyboss.atlassian.net/browse/PROD-1899
https://buddyboss.atlassian.net/browse/PROD-1920
https://buddyboss.atlassian.net/browse/PROD-1902
https://buddyboss.atlassian.net/browse/PROD-2127
https://buddyboss.atlassian.net/browse/PROD-1922
https://buddyboss.atlassian.net/browse/PROD-1921
https://buddyboss.atlassian.net/browse/PROD-2106

### General Note
Keep all conversations related to this PR in the associated Jira issue(s). Do NOT add comment on this PR or edit this PR’s description.

### Notes to Developer
* Ensure the IDs (i.e. PROD-1) of all associated Jira issues are reference in this PR’s title
* Ensure that you have achieved the Definition of Done before submitting for review
* When this PR is ready for review, move the associate Jira issue(s) to “Needs Review” (or “Code Review” for Dev Tasks)

### Notes to Reviewer
* Ensure that the Definition of Done have been achieved before approving a PR
* When this PR is approved, move the associated Jira issue(s) to “Needs QA” (or “Approved” for Dev Tasks)

### CODE REVIEW

File: src/bp-core/admin/bp-core-admin-functions.php [hunk 1]
```diff
@@ -2179,47 +2177,48 @@ function bp_save_member_type_post_metabox_data( $post_id ) {
...
+	// verify nonce.
+	// phpcs:ignore WordPress.Security.ValidatedSanitizedInput.InputNotValidated, WordPress.Security.ValidatedSanitizedInput.MissingUnslash, WordPress.Security.ValidatedSanitizedInput.InputNotSanitized
 	if ( ! wp_verify_nonce( $_POST['_bp-member-type-nonce'], 'bp-member-type-edit-member-type' ) ) {
 		return;
 	}
 
-	// Save data
-	$data = isset( $_POST['bp-member-type'] ) ? $_POST['bp-member-type'] : array();
+	// Save data.
+	$data = isset( $_POST['bp-member-type'] ) ? $_POST['bp-member-type'] : array(); // phpcs:ignore WordPress.Security.ValidatedSanitizedInput.InputNotValidated, WordPress.Security.ValidatedSanitizedInput.MissingUnslash, WordPress.Security.ValidatedSanitizedInput.InputNotSanitized
```
Comment (chetansatasiya) [inline src/bp-core/admin/bp-core-admin-functions.php:2195]: don't make the warning silence. Use the sanitizing functions

File: src/bp-core/admin/bp-core-admin-functions.php [hunk 2]
```diff
@@ -2232,10 +2231,10 @@ function bp_save_member_type_post_metabox_data( $post_id ) {
...
-	$enable_group_type_enable_invite = isset( $_POST['bp-member-type-enabled-invite'] ) ? $_POST['bp-member-type-enabled-invite'] : '';
+	$enable_group_type_create        = isset( $_POST['bp-group-type'] ) ? $_POST['bp-group-type'] : ''; // phpcs:ignore WordPress.Security.ValidatedSanitizedInput.InputNotSanitized, WordPress.Security.ValidatedSanitizedInput.MissingUnslash
+	$enable_group_type_auto_join     = isset( $_POST['bp-group-type-auto-join'] ) ? $_POST['bp-group-type-auto-join'] : ''; // phpcs:ignore WordPress.Security.ValidatedSanitizedInput.InputNotSanitized, WordPress.Security.ValidatedSanitizedInput.MissingUnslash
+	$enable_group_type_invite        = isset( $_POST['bp-member-type-invite'] ) ? $_POST['bp-member-type-invite'] : ''; // phpcs:ignore WordPress.Security.ValidatedSanitizedInput.InputNotSanitized, WordPress.Security.ValidatedSanitizedInput.MissingUnslash
+	$enable_group_type_enable_invite = isset( $_POST['bp-member-type-enabled-invite'] ) ? $_POST['bp-member-type-enabled-invite'] : ''; // phpcs:ignore WordPress.Security.ValidatedSanitizedInput.InputNotSanitized, WordPress.Security.ValidatedSanitizedInput.MissingUnslash
```
Comment (chetansatasiya) [inline src/bp-core/admin/bp-core-admin-functions.php:2237]: don't make the warning silence. Use the sanitizing functions

**Review by chetansatasiya — CHANGES_REQUESTED** [review]
Follow the PHPCS properly and remove the all the PHPCS silence comments.

File: src/bp-templates/bp-nouveau/buddypress/groups/groups-loop.php [hunk 3]
```diff
@@ -6,91 +6,163 @@
...
 	</ul>
 
+	<!-- Leave Group confirmation popup -->
+	<div class="bb-leave-group-popup bb-action-popup" style="display: none">
+		<transition name="modal">
+			<div class="modal-mask bb-white bbm-model-wrap">
+				<div class="modal-wrapper">
+					<div class="modal-container">
+						<header class="bb-model-header">
+							<h4><span class="target_name"><?php echo esc_html__( 'Leave Group', 'buddyboss' ); ?></span></h4>
```
Comment (KartikSuthar) [inline src/bp-templates/bp-nouveau/buddypress/groups/groups-loop.php:147]:
```suggestion
							<h4><span class="target_name"><?php esc_html_e( 'Leave Group', 'buddyboss' ); ?></span></h4>
```

File: src/bp-templates/bp-nouveau/buddypress/groups/groups-loop.php [hunk 4]
```diff
@@ -6,91 +6,163 @@
...
+							<a class="bb-close-leave-group bb-model-close-button" href="#">
+								<span class="bb-icon bb-icon-close"></span>
+							</a>
+						</header>
+						<div class="bb-leave-group-content bb-action-popup-content">
+							<p><?php echo esc_html__( 'Are you sure you want to leave ', 'buddyboss' ); ?><span class="bb-group-name"></span></p>
```
Comment (KartikSuthar) [inline src/bp-templates/bp-nouveau/buddypress/groups/groups-loop.php:153]:
```suggestion
							<p><?php esc_html_e( 'Are you sure you want to leave ', 'buddyboss' ); ?><span class="bb-group-name"></span></p>
```

File: src/bp-templates/bp-nouveau/buddypress/groups/groups-loop.php [hunk 5]
```diff
@@ -6,91 +6,163 @@
...
+						</div>
+						<footer class="bb-model-footer flex align-items-center">
+							<a class="bb-close-leave-group bb-close-action-popup" href="#"><?php echo esc_html__( 'Cancel', 'buddyboss' ); ?></a>
```
Comment (KartikSuthar) [inline src/bp-templates/bp-nouveau/buddypress/groups/groups-loop.php:156]:
```suggestion
							<a class="bb-close-leave-group bb-close-action-popup" href="#"><?php echo esc_html_e( 'Cancel', 'buddyboss' ); ?></a>
```

File: src/bp-templates/bp-nouveau/buddypress/groups/groups-loop.php [hunk 6]
```diff
@@ -6,91 +6,163 @@
...
+							<a class="button push-right bb-confirm-leave-group" href="#"><?php echo esc_html__( 'Confirm', 'buddyboss' ); ?></a>
```
Comment (KartikSuthar) [inline src/bp-templates/bp-nouveau/buddypress/groups/groups-loop.php:157]:
```suggestion
							<a class="button push-right bb-confirm-leave-group" href="#"><?php esc_html_e( 'Confirm', 'buddyboss' ); ?></a>
```

File: src/bp-templates/bp-nouveau/buddypress/groups/single/cover-image-header.php [hunk 7]
```diff
@@ -80,38 +87,132 @@
...
+					</div><!-- .item-meta -->
+				<?php endif; ?>
 
-						<?php echo bp_nouveau_group_meta()->extra; ?>
+				<?php if ( bb_platform_group_headers_element_enable( 'group-activity' ) ) : ?>
+				<p class="last-activity item-meta">
+					<?php
+					printf(
+						/* translators: %s = last activity timestamp (e.g. "active 1 hour ago") */
+						esc_attr__( 'Active %s', 'buddyboss' ),
```
Comment (KartikSuthar) [inline src/bp-templates/bp-nouveau/buddypress/groups/single/cover-image-header.php:127]:
```suggestion
						esc_html__( 'Active %s', 'buddyboss' ),
```

**Review by KartikSuthar — CHANGES_REQUESTED** [review]

File: src/bp-core/bp-core-update.php [hunk 8]
```diff
@@ -1686,3 +1700,63 @@ function bb_update_to_1_9_1() {
...
+	}
+
+	// Get BuddyBoss theme options.
+	global $buddyboss_theme_options;
+
+	// Get BuddyBoss theme version.
+	$bb_theme_version = wp_get_theme()->get( 'Version' );
+
+	// Check the theme already upto date or not.
+	if ( function_exists( 'buddyboss_theme' ) && version_compare( $bb_theme_version, '1.8.7', '>=' ) ) {
```
[the remark this answers is not in the document]
Comment (milanhirapra) [reply src/bp-core/bp-core-update.php:1725]: Its release in 1.8.7.

File: src/bp-templates/bp-nouveau/buddypress/groups/groups-loop.php [hunk 9]
```diff
@@ -6,91 +6,165 @@
...
+$meta_group_type    = ! bb_platform_group_element_enable( 'group-type' ) ? 'meta-group-type-hidden' : '';
+$group_members      = ! bb_platform_group_element_enable( 'members' ) ? 'group-members-hidden' : '';
+$join_button        = ! bb_platform_group_element_enable( 'join-buttons' ) ? 'group-join-button-hidden' : '';
+$group_alignment    = bb_platform_group_grid_style( 'left' );
+$group_cover_height = function_exists( 'bb_get_group_cover_image_height' ) ? bb_get_group_cover_image_height() : 'small';
+
+$group_cover_height_class = 'cover-large';
+if ( 'small' === $group_cover_height ) {
+	$group_cover_height_class = '';
+}
```
Comment (KartikSuthar) [inline src/bp-templates/bp-nouveau/buddypress/groups/groups-loop.php:35]: Remove this code, add corectly here: https://github.com/buddyboss/buddyboss-platform/pull/3339/files#diff-da4134c9b4d6993bbb5f8450b6676f905ba0e0b59de3ba349c62ddd6f8a46a0eR64
`'cover-' . $group_cover_height`

File: src/bp-templates/bp-nouveau/buddypress/groups/groups-loop.php [hunk 10]
```diff
@@ -6,91 +6,165 @@
...
+							$group_cover_image_url = bp_attachments_get_attachment(
+								'url',
+								array(
+									'object_dir' => 'groups',
+									'item_id'    => bp_get_group_id(),
+								)
+							);
+							$has_default_cover     = function_exists( 'bb_attachment_get_cover_image_class' ) ? bb_attachment_get_cover_image_class( bp_get_group_id(), 'group' ) : '';
+							?>
+							<div class="bs-group-cover only-grid-view <?php echo esc_attr( $has_default_cover . ' ' . $group_cover_height_class ); ?>"><a href="<?php bp_group_permalink(); ?>"><img src="<?php echo esc_url( $group_cover_image_url ); ?>"></a></div>
```
Comment (KartikSuthar) [inline src/bp-templates/bp-nouveau/buddypress/groups/groups-loop.php:64]: follow this here: https://github.com/buddyboss/buddyboss-platform/pull/3339/files#r817572973

File: src/bp-templates/bp-nouveau/buddypress/groups/groups-loop.php [hunk 11]
```diff
@@ -6,91 +6,165 @@
...
 
 					<?php if ( ! bp_disable_group_avatar_uploads() ) : ?>
-						<div class="item-avatar">
-							<a href="<?php bp_group_permalink(); ?>"><?php bp_group_avatar( bp_nouveau_avatar_args() ); ?></a>
-						</div>
+						<?php if ( bb_platform_group_element_enable( 'avatars' ) ) { ?>
+							<div class="item-avatar">
+								<a href="<?php bp_group_permalink(); ?>" class="group-avatar-wrap"><?php bp_group_avatar( bp_nouveau_avatar_args() ); ?></a>
+							</div>
+						<?php } ?>
```
Comment (KartikSuthar) [inline src/bp-templates/bp-nouveau/buddypress/groups/groups-loop.php:76]: combine 2 if into one with above one.

File: src/bp-templates/bp-nouveau/buddypress/groups/single/cover-image-header.php [hunk 12]
```diff
@@ -80,38 +85,137 @@
 		<?php if ( ! bp_nouveau_groups_front_page_description() ) : ?>
 			<div id="item-header-content">
 
-				<p class="highlight group-status bp-tooltip" data-bp-tooltip-length="large" data-bp-tooltip-pos="up" data-bp-tooltip="<?php echo esc_html( bp_get_group_status_description() ); ?>"><strong><?php echo wp_kses( bp_nouveau_group_meta()->status, array( 'span' => array( 'class' => array() ) ) ); ?></strong></p>
-				<p class="activity">
-					<a href="<?php echo esc_url( bp_get_group_permalink() . 'members' ); ?>"><?php echo esc_html( bp_get_group_member_count() ); ?></a>
-				</p>
+				<?php if ( function_exists( 'bp_enable_group_hierarchies' ) && bp_enable_group_hierarchies() ) : ?>
+					<?php
```
Comment (KartikSuthar) [inline src/bp-templates/bp-nouveau/buddypress/groups/single/cover-image-header.php:89]:
```suggestion
				<?php if ( function_exists( 'bp_enable_group_hierarchies' ) && bp_enable_group_hierarchies() ) : 
```

File: src/bp-templates/bp-nouveau/buddypress/groups/groups-loop.php [hunk 13]
```diff
@@ -6,91 +6,159 @@
...
+						<div class="group-item-wrap">
 
-						<?php if ( bp_nouveau_group_has_meta() ) : ?>
+							<div class="item-block">
 
-							<p class="item-meta group-details"><?php bp_nouveau_group_meta(); ?></p>
+								<h2 class="list-title groups-title"><?php bp_group_link(); ?></h2>
 
-						<?php endif; ?>
+								<div class="item-meta-wrap <?php echo bb_platform_group_element_enable( 'last-activity' ) || empty( $meta_privacy ) || empty( $meta_group_type ) ? 'has-meta' : 'meta-hidden'; ?> ">
```
Comment (chetansatasiya) [inline src/bp-templates/bp-nouveau/buddypress/groups/groups-loop.php:81]: Need escaping functions

**Review by chetansatasiya — DISMISSED** [review]

**Review by KartikSuthar — APPROVED** [review]

File: src/bp-templates/bp-nouveau/buddypress/members/single/cover-image-header.php [hunk 14]
```diff
@@ -27,99 +28,244 @@
...
+$is_enabled_last_active      = bb_enabled_profile_header_layout_element( 'last-active' );
+$is_enabled_followers        = bb_enabled_profile_header_layout_element( 'followers' );
+$is_enabled_following        = bb_enabled_profile_header_layout_element( 'following' );
+$is_enabled_social_networks  = bb_enabled_profile_header_layout_element( 'social-networks' ) && bp_member_type_enable_disable() && function_exists( 'bb_enabled_member_social_networks' ) && bb_enabled_member_social_networks();
 
-		<?php } ?>
-	</div>
+$my_profile 					=  '';
+$user_social_networks_urls      = '';
+$social_networks_urls_div_class = 'social-networks-hide';
```
Comment (KartikSuthar) [inline src/bp-templates/bp-nouveau/buddypress/members/single/cover-image-header.php:44]: Fix alignment here.

File: src/bp-templates/bp-nouveau/buddypress/members/single/member-header.php [hunk 15]
```diff
@@ -7,51 +7,213 @@
...
+	$is_enabled_member_handle    = bb_enabled_profile_header_layout_element( 'member-handle' );
+	$is_enabled_joined_date      = bb_enabled_profile_header_layout_element( 'joined-date' );
+	$is_enabled_last_active      = bb_enabled_profile_header_layout_element( 'last-active' );
+	$is_enabled_followers        = bb_enabled_profile_header_layout_element( 'followers' );
+	$is_enabled_following        = bb_enabled_profile_header_layout_element( 'following' );
+	$is_enabled_social_networks  = bb_enabled_profile_header_layout_element( 'social-networks' ) && bp_member_type_enable_disable() && function_exists( 'bb_enabled_member_social_networks' ) && bb_enabled_member_social_networks();
+
+	$my_profile 					=  '';
+	$user_social_networks_urls      = '';
+	$social_networks_urls_div_class = 'social-networks-hide';
```
Comment (KartikSuthar) [inline src/bp-templates/bp-nouveau/buddypress/members/single/member-header.php:27]: Fix alignment here.

**Review by KartikSuthar — APPROVED** [review]

**Pull request merged**